.github/workflows/ci-build.yml

```diff
@@ -0,0 +1,57 @@
+
+name: CI
+
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+   
+jobs:
+  Code-Quality-Checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: 
+        scripts:
+          - run-linter
+          - check-docstyle
+          - detect-dead-code
+          - detect-common-errors
+          - measure-cyclomatic-complexity
+          - measure-maintainability-index    
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.6'
+    - name: ${{ matrix.scripts }}
+      run: bash qa/${{ matrix.scripts }}.sh;
+  Unit-Tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.6'
+    - name: Install virtualenv
+      run: pip install -U pip && pip install virtualenv
+    - name: Run Unit Tests
+      run: bash qa/runtests.sh
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        file: ./coverage.xml
+    - name: Install oc
+      uses: redhat-actions/oc-installer@v1
+      with:
+        version: '4.6'
+    - name: Run oc process
+      run: oc process --local -f openshift/template.yaml -o yaml > blueprint.yaml
```

Comment:
How this works when the template needs mandatory param?

Reply:
https://github.com/instrumenta/kubeval/issues/33 I was just trying out a solution , can you please tell me what mandatory param ?

Reply:
The solution works : https://github.com/sawood14012/f8a-stacks-report/runs/1565527998?check_suite_focus=true

Reply:
Mandatory param is one which should to passed while applying template. i.e using -p to `oc process`.

Reply:
process command just provides us list of objects right , anyways we can add and override parameters , i found this in the documentation 
"You can override any parameter values defined in the file by adding the -p option for each <name>=<value> pair you want to override. A parameter reference may appear in any text field inside the template items." 
https://docs.openshift.com/container-platform/3.6/dev_guide/templates.html

Reply:
sorry for late replies i was on a leave today

Reply:
We can merge this once you convert the yaml validation as separate job.